tests/api/test_checkout.py

```diff
@@ -1776,3 +1778,205 @@ def test_is_fully_paid_no_payment(checkout_with_item):
+        update_checkout_private_meta, checkout_private_meta_update_variables
+    )
+    content = get_graphql_content(response)
+    errors = content["data"]["checkoutUpdatePrivateMetadata"]["errors"]
+    assert len(errors) == 0
+    checkout.refresh_from_db()
+    assert checkout.get_private_meta(namespace="test", client="client1") == {"foo": "bar"}
+
+
+@pytest.fixture
+def clear_checkout_meta():
+    return """
+        mutation checkoutClearMetadata($token: UUID!, $input: MetaPath!) {
+            checkoutClearMetadata(token: $token, input: $input) {
+                errors {
+                    message
+                }
+            }
+        }
+    """
+
+
+@pytest.fixture
+def clear_checkout_private_meta():
+    return """
+        mutation checkoutClearPrivateMetadata($id: ID!, $input: MetaPath!) {
+            checkoutClearPrivateMetadata(id: $id, input: $input) {
+                errors {
+                    message
+                }
+            }
+        }
+    """
+
+
+@pytest.fixture
+def clear_meta_variables(checkout):
+    return {
+        "token": checkout.token,
+        "input": {"namespace": "test", "clientName": "client1", "key": "foo"},
+    }
+
+
+@pytest.fixture
+def clear_meta_private_variables(checkout):
+    return {
+        "id": checkout.pk,
+        "input": {
+            "namespace": "testPrivate",
+            "clientName": "clientPrivate1",
+            "key": "foo",
+        },
+    }
+
+
+@pytest.fixture
+def checkout_with_meta(checkout):
+    checkout.store_meta(namespace="test", client="client1", item={"foo": "bar"})
+    checkout.store_private_meta(
+        namespace="testPrivate", client="clientPrivate1", item={"foo": "bar"}
+    )
+    checkout.save()
+    return checkout
+
+
+def test_user_without_permission_cannot_clear_private_meta(
+    staff_user, staff_api_client, clear_checkout_private_meta, clear_meta_private_variables
+):
+    assert not staff_user.has_perm(CheckoutPermissions.MANAGE_CHECKOUTS)
+    response = staff_api_client.post_graphql(
+        clear_checkout_private_meta, clear_meta_private_variables
+    )
+    assert_no_permission(response)
+
+
+def test_user_with_permission_can_clear_meta(
+    staff_user,
+    staff_api_client,
+    clear_checkout_meta,
+    clear_meta_variables,
+    checkout_with_meta,
+    permission_manage_checkouts,
+):
+    staff_user.user_permissions.add(permission_manage_checkouts)
+    assert staff_user.has_perm(CheckoutPermissions.MANAGE_CHECKOUTS)
+    response = staff_api_client.post_graphql(clear_checkout_meta, clear_meta_variables)
+    assert response.status_code == 200
+    content = get_graphql_content(response)
+    errors = content["data"]["checkoutClearMetadata"]["errors"]
+    assert len(errors) == 0
+    checkout_with_meta.refresh_from_db()
+    current_meta = checkout_with_meta.get_meta(namespace="test", client="client1")
+    assert current_meta == {}
+
+
+def test_user_with_permission_can_clear_private_meta(
```

Comment:
`staff_user`

Reply:
@EwanValentine it fails because `clear_meta_private_variables` has `id` instead of `token`

Reply:
from the error that you provided:
```
clear_meta_private_variables = {'id': UUID('d38e85ce-4eb6-4d72-816e-4f107cd564dd')
```

Reply:
@korycins ah my bad, my assumption was private meta was queried by the global id still? Whereas the public meta was queried by the token? I think the schema reflects this as well currently

Reply:
Ah I see, so in `test_order` we're doing: `order_id = graphene.Node.to_global_id("Order", order.id)` which I'm missing, I removed this initially because `id` was missing, so I think I just need to change that to `pk`